visualization/app/codeCharta/util/reduxHelper.ts

```diff
@@ -16,6 +17,10 @@ export function addItemToArray(array: any[], item: any): any[] {
 	return array
 }
 
+export function isActionOfType(actionType: string, actions) {
+	return Object.values(actions).includes(actionType)
```

Comment:
This seems very costly. These actions are all string based enums. AFAIK they all have a value that is identical to their key. It should therefore suffice to do:

 ```suggestion
	return actions[actionType]
```

This is a whole different complexity level: from `O(n)` to `O(1)`. I also wonder why these enums have to be string based? It would be more efficient to use number based enums.

Reply:
https://redux.js.org/basics/actions/
Redux actions are strings.

I will apply your suggestion though. As stated before: keys and values are the same. Hopefully this won't change.